hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/NodeDecommissionMetrics.java

```diff
@@ -0,0 +1,245 @@
+   */
+  @Override
+  public void getMetrics(MetricsCollector collector, boolean all) {
+    MetricsRecordBuilder builder = collector
+        .addRecord(METRICS_SOURCE_NAME);
+    trackedDecommissioningMaintenanceNodesTotal.snapshot(builder, all);
+    trackedRecommissionNodesTotal.snapshot(builder, all);
+    trackedPipelinesWaitingToCloseTotal.snapshot(builder, all);
+    trackedContainersUnderReplicatedTotal.snapshot(builder, all);
+    trackedContainersUnhealthyTotal.snapshot(builder, all);
+    trackedContainersSufficientlyReplicatedTotal.snapshot(builder, all);
+
+    registry.snapshot(builder, all);
+  }
+
+  /**
+   * Unregister the metrics instance.
+   */
+  public void unRegister() {
+    DefaultMetricsSystem.instance().unregisterSource(METRICS_SOURCE_NAME);
+  }
+
+  public void setTrackedDecommissioningMaintenanceNodesTotal(
+            long numNodesTracked) {
+    trackedDecommissioningMaintenanceNodesTotal
+        .set(numNodesTracked);
+  }
+
+  public void setTrackedRecommissionNodesTotal(
+          long numNodesTrackedRecommissioned) {
+    trackedRecommissionNodesTotal.set(numNodesTrackedRecommissioned);
+  }
+
+  public void setTrackedPipelinesWaitingToCloseTotal(
+          long numTrackedPipelinesWaitToClose) {
+    trackedPipelinesWaitingToCloseTotal
+        .set(numTrackedPipelinesWaitToClose);
+  }
+
+  public void setTrackedContainersUnderReplicatedTotal(
+          long numTrackedUnderReplicated) {
+    trackedContainersUnderReplicatedTotal
+        .set(numTrackedUnderReplicated);
+  }
+
+  public void setTrackedContainersUnhealthyTotal(
+          long numTrackedUnhealthy) {
+    trackedContainersUnhealthyTotal
+        .set(numTrackedUnhealthy);
+  }
+
+  public void setTrackedContainersSufficientlyReplicatedTotal(
+          long numTrackedSufficientlyReplicated) {
+    trackedContainersSufficientlyReplicatedTotal
+        .set(numTrackedSufficientlyReplicated);
+  }
+
+  public long getTrackedDecommissioningMaintenanceNodesTotal() {
+    return trackedDecommissioningMaintenanceNodesTotal.value();
+  }
+
+  public long getTrackedRecommissionNodesTotal() {
+    return trackedRecommissionNodesTotal.value();
+  }
+
+  public long getTrackedPipelinesWaitingToCloseTotal() {
+    return trackedPipelinesWaitingToCloseTotal.value();
+  }
+
+  public long getTrackedContainersUnderReplicatedTotal() {
+    return trackedContainersUnderReplicatedTotal.value();
+  }
+
+  public long getTrackedContainersUnhealthyTotal() {
+    return trackedContainersUnhealthyTotal.value();
+  }
+
+  public long getTrackedContainersSufficientlyReplicatedTotal() {
+    return trackedContainersSufficientlyReplicatedTotal.value();
+  }
+
+  public void metricRecordPipelineWaitingToCloseByHost(String host,
+                                                       long num) {
+    trackedPipelinesWaitingToCloseByHost.computeIfAbsent(host,
+        hostID -> registry.newGauge(
+            Interns.info("trackedPipelinesWaitingToClose-" + hostID,
```

Comment:
This block is repeated 4 times with just the name and value changing:

```
Interns.info("trackedSufficientlyReplicated-" + hostID,
                "Number of sufficiently replicated containers "
                    + "for host in decommissioning and "
                    + "maintenance mode"), 0L)).set(sufficientlyReplicated);
```

Could you move it into a private method to reduce the duplicated code?

Reply:
Thanks.  Moved to new private method : 
`  private TrackedWorkflowContainerState createContainerMetricsInfo`